tests/test_extension_registry.py

```
@@ -0,0 +1,1632 @@
+        self.assertEqual(categories[0]['description'], 'Updated Description')
+        self.assertEqual(categories[0]['sort_order'], 2)
+        
+        extensions = self.manager.get_extensions()
+        self.assertEqual(extensions[0]['description'], 'Updated Extension')
+    
+    def test_import_partial_overwrite(self):
+        """Test import with partial overlap - only matching entities should be overwritten."""
+        # Create initial data
+        cat1_id = self.manager.create_category("Category 1", "Original 1", 1, True)
+        cat2_id = self.manager.create_category("Category 2", "Original 2", 2, True)
+        ext1_id = self.manager.create_extension(".ext1", cat1_id, "Extension 1", None, True, False, True, False, False)
```

Ruff (0.12.2)

Local variable `ext1_id` is assigned to but never used

Remove assignment to unused variable `ext1_id`

(F841)